clang/lib/Sema/SemaOverload.cpp

```diff
@@ -1536,10 +1536,19 @@ static bool IsOverloadOrOverrideImpl(Sema &SemaRef, FunctionDecl *New,
                !F->isExplicitObjectMemberFunction();
       };
 
-      if (IsImplicitWithNoRefQual(Old) != IsImplicitWithNoRefQual(New) &&
-          CompareType(OldObjectType.getNonReferenceType(),
-                      NewObjectType.getNonReferenceType()))
-        return true;
+      if ((Old->getRefQualifier() != RQ_None ||
```

Comment:
Perhaps too early in the morning, but this whole predicate (and the one below it) are both... rough :)  Is there a good way to make them more readable by extracting named variables or something?